packages/vite/src/node/http.ts

```diff
@@ -98,6 +98,11 @@ export async function resolveHttpServer(
     return require('http2').createSecureServer(
       {
         ...httpsOptions,
+
+        // Manually increase the session memory to prevent 502 ENHANCE_YOUR_CALM
+        // errors on large numbers of requests
+        maxSessionMemory: (httpsOptions as any).maxSessionMemory ?? 1000,
+
```

Comment:
We can simplify the implementation:
```suggestion
        // Manually increase the session memory to prevent 502 ENHANCE_YOUR_CALM
        // errors on large numbers of requests
        maxSessionMemory: 1000,
        ...httpsOptions,
```